src/main/topLevelPages/myDashboard/newVirtualContributorWizard/useNewVirtualContributorWizard.tsx

```diff
@@ -496,67 +500,75 @@ const useNewVirtualContributorWizard = (): useNewVirtualContributorWizardProvide
       setStep('insufficientPrivileges');
       return false;
     }
+    const createdVc = await executeMutation({ values, accountId });
 
-    try {
-      const { data } = await createVirtualContributor({
-        variables: {
-          virtualContributorData: {
-            accountID: accountId,
-            profileData: {
-              displayName: values.name,
-              tagline: values.tagline,
-              description:
-                values.description ?? t('createVirtualContributorWizard.createdVirtualContributor.description'),
-            },
-            aiPersona: {
-              aiPersonaService: {
-                bodyOfKnowledgeID: vcBoKId,
-              },
-            },
-          },
-        },
-      });
+    const virtualContributorId = createdVc?.id;
 
-      const virtualContributorId = data?.createVirtualContributor.id;
-
-      if (virtualContributorId) {
-        if (parentRoleSetId) {
-          // the VC cannot be added to the BoK community
-          // if it's not part of the parent community
-          await addVirtualContributorToRole({
-            variables: {
-              roleSetId: parentRoleSetId,
-              contributorId: virtualContributorId,
-              role: CommunityRoleType.Member,
-            },
-          });
-        }
-
-        // add the VC to the BoK community
+    if (virtualContributorId) {
+      if (parentRoleSetId) {
+        // the VC cannot be added to the BoK community
+        // if it's not part of the parent community
         await addVirtualContributorToRole({
           variables: {
-            roleSetId: roleSetId,
+            roleSetId: parentRoleSetId,
             contributorId: virtualContributorId,
             role: CommunityRoleType.Member,
           },
         });
+      }
 
-        // add vc's nameId to the cache for the TryVC dialog
-        if (data?.createVirtualContributor.nameID) {
-          addVCCreationCache(data?.createVirtualContributor.nameID);
-        }
+      // add the VC to the BoK community
+      await addVirtualContributorToRole({
+        variables: {
+          roleSetId: roleSetId,
+          contributorId: virtualContributorId,
+          role: CommunityRoleType.Member,
+        },
+      });
+
+      notify(
+        t('createVirtualContributorWizard.createdVirtualContributor.successMessage', { name: values.name }),
+        'success'
+      );
+
+      return true;
+    }
+
+    return false;
+  };
 
-        notify(
-          t('createVirtualContributorWizard.createdVirtualContributor.successMessage', { name: values.name }),
-          'success'
-        );
+  const executeMutation = async ({ values, accountId }: { values: VirtualContributorFromProps; accountId: string }) => {
```

Comment:
Let's rename the executeMutation to something more contextual e.g. createExternalVirtualContrilbutor

Reply:
This function is used for both `handleCreateVirtualContributor` and `handleCreateExternal` so this is the generic one. I'd rename `handleCreateVirtualContributor` to `handleCreate` as the base one and in my opinion the `VirtualContributor` part is redundant as we are in the context of `newVCWizard`.

Reply:
Ok but again mutation is too common. Let's include 'create' as it's not any mutation, it creates a VC.